fuel-vm/src/interpreter/memory.rs

```diff
@@ -414,24 +414,24 @@ pub(crate) fn memcopy(
     let (bc, of) = b.overflowing_add(c);
     let overflow = overflow || of;
 
+    let dst_before_src = a <= b;
     let range = MemoryRange::new(a, c);
     if overflow
         || ac > VM_MAX_RAM
         || bc > VM_MAX_RAM
         || c > MEM_MAX_ACCESS_SIZE
-        || a <= b && b < ac
+        || dst_before_src && b < ac
         || b <= a && a < bc
```

Comment:
Hmm, also it seems that we don't catch `a <= bc && bc < ac` or `b <= ac && ac < bc`